pg_analytics/src/hooks/executor.rs

```diff
@@ -49,28 +50,35 @@ pub fn executor_run(
         }
 
         // Parse the query into a LogicalPlan
-        let logical_plan = pg_plan.get_logical_plan(&query);
+        let LogicalPlanDetails {
```

Comment:
@neilyio would you recommend a more idiomatic way to do this or is fine to you?

I would prefer something like 

```rust
impl LogicalPlanDetails {
   pub fn logical_plan() -> LogicalPlan { ... }
   pub fn includes_udf() -> bool { ... }
}
```

Reply:
I agree with you, it’s more common to use access methods in Rust. Public struct fields, I think, should only be used if it would be awkward to initialize the struct with builder methods.